src/provisioner/provisioner.py

```diff
@@ -244,6 +244,34 @@ def _provision_components(cortx_conf: MappedConf, interfaces: Enum, apply_phase:
 
     @staticmethod
     def cluster_bootstrap(cortx_conf_url: str, force_override: bool = False):
+        """
+        Description:
+        Configures Cluster Components
+        1. Compares current installed version with New version
+        2. Invoke Mini Provisioners of cluster components deploy/upgrade based on version compatibility
+        Paramaters:
+        [IN] CORTX Config URL
+        """
+        cortx_conf = MappedConf(cortx_conf_url)
+        node_id = Conf.machine_id
+        installed_version = cortx_conf.get(f'node>{node_id}>provisioning>version')
+        release_version = CortxProvisioner.cortx_release.get_release_version()
+        if installed_version is None:
+            CortxProvisioner.cluster_deploy(cortx_conf_url,force_override)
+        else:
+            ret_code = CortxProvisioner.cortx_release.version_check(
```

Comment:
We need to handle a case where release_version > installed_version but is not compatible and an error will be shown accordingly.

Reply:
@ujjwalpl , Kailash is already working on writing this logic in utils for adding actually compatible versions check. Once that is implemented we can add in provisioner also, For now I will mark it as TODO.